README.adoc

```diff
@@ -13,109 +15,206 @@ Use this action to inform CloudBees Platform that an artifact version was create
-
 | `name`
 | String
 | Yes
-| The name of the artifact to send to the {PRODUCT} for artifact traceability purposes.
+| The name of the artifact reported to the CloudBees platform.
 
 | `url`
 | String
 | Yes
-| The URL where the artifact version is located.  For example, docker.io/myapp/myimg:1.0.0.
+| The URL where the artifact version is located.
+For example, `docker.io/myapp/myimg:1.0.0`.
 
 | `version`
 | String
 | Yes
-| The version of the artifact to send to the {PRODUCT} for artifact traceability purposes.
-
-| `repository-url`
-| String
-| No
-| The clone URL of the repository.
+| The version of the artifact reported to the CloudBees platform.
 
 | `commit`
 | String
-| No
+| Only required if a different repository/branch from the workflow is checked out.
 | The commit ID from the source repository.
+
 | `commit-url`
 | String
-| No
+| Only required if a different repository/branch from the workflow is checked out.
 | The commit URL from the source repository.
 
 | `ref`
 | String
+| Only required if a different repository/branch from the workflow run is checked out.
+| The ref or branch of the artifact repository.
+
+| `repository-url`
+| String
+| Only required if a different repository/branch from the workflow run is checked out.
+| The artifact repository URL.
+
+| `digest`
+| String
 | No
-| The ref or branch of the checked-out repository.
-|===
+| The hash or checksum that uniquely identifies the artifact version.
+
+| `labels`
+| String
+| No
+| A comma-separated list of artifact labels.
 
-Note: By default `repository-url`, `commit`, `commit-url` and `ref` inputs
-are not required unless checking out a different repository/branch
-from the workflow. See the second usage example below to use the outputs
-from the checkout action.
+|===
 
 == Outputs
 
-[cols="2a,2a,5a",options="header"]
+[cols="2a,1a,3a",options="header"]
 .Output details
 |===
-
 | Output name
 | Data type
 | Description
 
 | `artifact-id`
 | String
-| The unique identifier of the artifact.
-
+| The unique identifier of the artifact reported to the CloudBees platform.
 |===
 
-== Usage example
 
-In your YAML file using the current workflow/branch, add:
+== Usage examples
+
+=== Basic example
+
+The following is a basic example of using the action:
 
 [source,yaml]
 ----
-      - name: Inform system that an artifact version was created
-        uses: cloudbees-io/register-build-artifact@v1
-        with:
-          name: "myApp"
-          version: "1.0.0"
-          url: "docker.io/myapp/myimg:1.0.0"
-          digest: "sha256:5e0687d9471728b0a200e0a3840fa16466150c58f628dc6e6c40a9b07e3b3609"
-          labels: "Enter,     your labels,here!    , Thanks."
+jobs:
+  register_build_artifact:
+      uses: cloudbees-io/register-build-artifact@v1
+      with:
+        name: "myApp"
+        url: "docker.io/myapp/myimg:1.0.0"
+        version: "1.0.0"
+
 ----
 
-Using the output from the checkout action, add:
+=== Using the action output
```

Comment:
This example is using the output from `checkout` action as input in `register-build-artifact`. While this is useful, we could also add an example for using the `artifact-id` output from this step in another downstream action. We can use a simplified version of the step "Print output parameter artifact IDs from Kaniko action" from https://github.com/cloudbees-io/kaniko/blob/be758833aea762cf5512a3f0b80718b03cae5cd9/README.adoc#using-the-action-output